modules/component/src/core/cd-aware/strategy.ts

```diff
@@ -0,0 +1,330 @@
+
+/**
+ *
+ * Pessimistic1 Strategy
+ *
+ * This strategy is providing only Angular built-in features.
+ *  If Ivy is present it uses the latest methods as they are.
+ *
+ * - \>=ViewEngine
+ *
+ * |          | Render Method      | Coalescing | Coalesce Scope |
+ * | -------- | -------------------| ---------- | -------------- |
+ * | ZoneFull | cdRef.markForCheck | ❌         | None           |
+ * | ZoneLess | cdRef.detectChange | ✔️         | None           |
+ *
+ * - Ivy
+ *
+ * |          | Render Method      | Coalescing | Coalesce Scope |
+ * | -------- | -------------------| ---------- | -------------- |
+ * | ZoneFull | ɵmarkDirty         | ❌         | None           |
+ * | ZoneLess | ɵmarkDirty         | ❌         | None           |
+ */
+export function createPessimistic1Strategy<T>(
+  cfg: StrategyFactoryConfig
+): CdStrategy<T> {
+  const inIvy = isIvy();
+  const inZone = hasZone(cfg.ngZone);
+  const durationSelector = getSaveDurationSelector(cfg.ngZone);
+
+  function render() {
+    if (inZone && !inIvy) {
+      cfg.cdRef.markForCheck();
+    } else if (!inZone && !inIvy) {
+      cfg.cdRef.detectChanges();
+    } else {
+      markDirty(cfg.component);
+    }
+  }
+
+  const behaviour = (o$: Observable<T>): Observable<T> => {
+    return !inZone && !inIvy
+      ? o$
+          .pipe
+          // @Notice this part of the code is in the coalescing PR
+          // coalesce(durationSelector)
+          ()
+      : o$;
+  };
+
+  return {
+    behaviour: () => behaviour,
+    render,
+    name: 'pessimistic1',
+  };
+}
+
+/**
+ *  Pessimistic2 Strategy
+ *
+ * This strategy is providing only Angular built-in features and optimized behavior in the ZoneLess environment.
+ * If Ivy is present it uses the latest methods as they are.
+ *
+ * - \>=ViewEngine
+ *
+ *  |          | Render Method      | Coalescing | Coalesce Scope |
+ * | -------- | -------------------| ---------- | -------------- |
+ * | ZoneFull | cdRef.markForCheck | ❌         | None           |
+ * | ZoneLess | cdRef.detectChange | ✔️         | Component      |
+ *
+ * - Ivy
+ *
+ * |          | Render Method      | Coalescing | Coalesce Scope |
+ * | -------- | -------------------| ---------- | -------------- |
+ * | ZoneFull | ɵmarkDirty         | ❌         | None           |
+ * | ZoneLess | ɵmarkDirty         | ❌         | None           |
+ */
+export function createPessimistic2Strategy<T>(
+  cfg: StrategyFactoryConfig
+): CdStrategy<T> {
+  const inIvy = isIvy();
+  const inZone = hasZone(cfg.ngZone);
+  const durationSelector = getSaveDurationSelector(cfg.ngZone);
+  const coalesceConfig = { context: cfg.component as any };
+
+  function render() {
+    if (inZone && !inIvy) {
+      cfg.cdRef.markForCheck();
+    } else if (!inZone && !inIvy) {
+      cfg.cdRef.detectChanges();
+    } else {
+      markDirty(cfg.component);
+    }
+  }
+
+  const behaviour = (o$: Observable<T>): Observable<T> => {
+    return !inZone && !inIvy
+      ? o$
+          .pipe
```

Comment:
Unsure of the goal here, again. `o$.pipe()` should be the same as `o$`.

Reply:
the internals here are not finished as it needs the operators in PR https://github.com/ngrx/platform/pull/2456
I should have deleted it completely...

Reply:
Done